### PR TITLE
Add in Search Database action `solutions` output field Rapid& VulnDB plugin

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
## Proposed Changes
Add in Search Database action `solutions` output field

### Description
Add in Search Database action `solutions` output field
Ticket MC-259

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "rapid7/Rapid7 Vulnerability \u0026 Exploit Database:2.1.0. Step name: get_content\n",
    "meta": {},
    "output": {
      "content_result": {
        "alternate_ids": "CVE/2021-30517,DEBIAN/DSA-4917-1",
        "content_type": "vulnerability",
        "description": "\n    \u003cp\u003eDebian: CVE-2021-30517: chromium -- security update\u003c/p\u003e\n  ",
        "published_at": "2021-05-21T00:00:00.000Z",
        "references": "debian-upgrade-chromium",
        "severity": "4",
        "solutions": "debian-upgrade-chromium",
        "title": "Debian: CVE-2021-30517: chromium -- security update"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_vulndb:2.1.0 --debug run < tests/get_content.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Rapid7 Vulnerability \u0026 Exploit Database:2.1.0. Step name: search_db\n",
    "meta": {},
    "output": {
      "results_found": true,
      "search_results": [
        {
          "identifier": "debian-cve-2021-30517",
          "link": "https://vdb-kasf1i23nr1kl2j4.rapid7.com/v1/content/debian-cve-2021-30517",
          "published_at": "2021-05-21T00:00:00.000Z",
          "solutions": "debian-upgrade-chromium",
          "title": "Debian: CVE-2021-30517: chromium -- security update"
        },
        {
          "identifier": "suse-cve-2021-30517",
          "link": "https://vdb-kasf1i23nr1kl2j4.rapid7.com/v1/content/suse-cve-2021-30517",
          "published_at": "2021-05-16T00:00:00.000Z",
          "solutions": "suse-upgrade-chromedriver,suse-upgrade-chromedriver-debuginfo,suse-upgrade-chromium,suse-upgrade-chromium-debuginfo",
          "title": "SUSE: CVE-2021-30517: SUSE Linux Security Advisory"
        },
        {
          "identifier": "google-chrome-cve-2021-30517",
          "link": "https://vdb-kasf1i23nr1kl2j4.rapid7.com/v1/content/google-chrome-cve-2021-30517",
          "published_at": "2021-05-11T00:00:00.000Z",
          "solutions": "google-chrome-upgrade-latest",
          "title": "Google Chrome Vulnerability: CVE-2021-30517 Type Confusion in V8"
        }
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_vulndb:2.1.0 --debug run < tests/search_db.json
</summary>
</details>

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Executing validator UnapprovedKeywordsValidator
338
[*] Plugin failed validation! The following validation errors occurred:

Validator "CloudReadyValidator" failed!
        Cause: 'enable_cache' must set to 'false' for a plugin to be Cloud Ready.Please check this field in plugin.spec and try again.


----
[*] Total time elapsed: 2037.0140000000001ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
./komand_rapid7_vulndb/util/extract.py:1:1: F401 'insightconnect_plugin_runtime.exceptions.PluginException' imported but unused
./komand_rapid7_vulndb/util/extract.py:4:1: F403 'from komand_rapid7_vulndb.util.transform import *' used; unable to detect undefined names
./komand_rapid7_vulndb/util/extract.py:48:21: F405 'transform' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:50:25: F405 'pop_non_relevant_search_fields' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:51:25: F405 'generate_link_attr' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:52:25: F405 'normalize_published_field' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:125:17: F405 'flatten_data_field' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:126:17: F405 'normalize_published_field' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:127:17: F405 'pop_non_relevant_module_fields' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:128:18: F405 'serialize_fields' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:132:17: F405 'flatten_data_field' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:133:17: F405 'serialize_alternate_ids' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:134:17: F405 'normalize_published_field' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:135:17: F405 'pop_non_relevant_vuln_fields' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:136:18: F405 'serialize_fields' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
./komand_rapid7_vulndb/util/extract.py:140:16: F405 'transform' may be undefined, or defined from star imports: komand_rapid7_vulndb.util.transform
[FAIL] Fails flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
Run started:2021-05-24 20:25:09.887988

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 517
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

[*] Checking for typos with Misspell...
[SUCCESS] Passes Misspell check

```

<summary>
make validate
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Executing validator UnapprovedKeywordsValidator
338
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "CloudReadyValidator" failed!
        Cause: 'enable_cache' must set to 'false' for a plugin to be Cloud Ready.Please check this field in plugin.spec and try again.


----
[*] Total time elapsed: 16433.411ms

```

<summary>
icon-validate --all . 
</summary>
</details>

### CODE REVIEW

**Comment by r7-kszczepanskagorna** [comment]
In 'make validate' there is a problem with Cloudready validator ('enable_cache' must set to 'false'). Should I leave it or change this 'enable_cache' to fit validator and requirements for CloudReady plugins? :)

File: rapid7_vulndb/help.md [hunk 1]
```diff
@@ -142,6 +143,7 @@ _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 2.1.0 - Add in Search Database action `solutions` output field
```
Comment (jschipp-r7) [inline rapid7_vulndb/help.md:146]:
```suggestion
* 2.1.0 - Return list of vulnerability solutions in the Search Database action with new `solutions` output field
```

**Review by jschipp-r7 — DISMISSED** [review]

**Review by mrinehart-r7 — DISMISSED** [review]

**Comment by mrinehart-r7** [comment]
> In 'make validate' there is a problem with Cloudready validator ('enable_cache' must set to 'false'). Should I leave it or change this 'enable_cache' to fit validator and requirements for CloudReady plugins? :)

Hi @r7-kszczepanskagorna - I looked through the plugin and there doesn't seem to be any cache usage. We can safely set that to false to satisfy the validator :)

**Comment by mrinehart-r7** [comment]
> > > In 'make validate' there is a problem with Cloudready validator ('enable_cache' must set to 'false'). Should I leave it or change this 'enable_cache' to fit validator and requirements for CloudReady plugins? :)
> > 
> > 
> > Hi @r7-kszczepanskagorna - I looked through the plugin and there doesn't seem to be any cache usage. We can safely set that to false to satisfy the validator :)
> 
> Mike, we should get a ticket created to fix that in the validator

There's nothing wrong with the validator - it's working properly. Someone just set cache to true in the spec but never used the cache within the code.

**Comment by jschipp-r7** [comment]
> > > > In 'make validate' there is a problem with Cloudready validator ('enable_cache' must set to 'false'). Should I leave it or change this 'enable_cache' to fit validator and requirements for CloudReady plugins? :)
> > > 
> > > 
> > > Hi @r7-kszczepanskagorna - I looked through the plugin and there doesn't seem to be any cache usage. We can safely set that to false to satisfy the validator :)
> > 
> > 
> > Mike, we should get a ticket created to fix that in the validator
> 
> There's nothing wrong with the validator - it's working properly. Someone just set cache to true in the spec but never used the cache within the code.

Oh, thanks for the clarification!

**Review by mrinehart-r7 — DISMISSED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Review by tschmidtke-r7 — APPROVED** [review]
